src/SMBusAnalyzer.cpp

```diff
@@ -42,11 +42,30 @@ AnalyzerChannelData* SMBusAnalyzer::AdvanceAllTo( U64 toSample )
 
 AnalyzerChannelData* SMBusAnalyzer::GetNearestTransitionChannel()
 {
+    // if neither channel has more transitions, block for more data.
+    while( !mSMBDAT->DoMoreTransitionsExistInCurrentData() && !mSMBCLK->DoMoreTransitionsExistInCurrentData() )
+    {
+        // DoMoreTransitionsExistInCurrentData will block for up to 250ms waiting for more data, and it will trigger a thread exit if the
+        // capture ends and there is still no more data.
```

Comment:
The 250ms is an implementation detail. I'm not sure if we want to list that specifically.